library/src/test/java/com/novoda/downloadmanager/NetworkFileDownloaderTest.java

```diff
@@ -77,11 +86,13 @@ public void emitsDownloadFinished_whenResponseCodeIsInvalid() throws IOException
 
     @Test
     public void emitsError_whenRequestExecutionFails() throws IOException {
-        given(httpClient.execute(requestCreator.createDownloadRequest(ANY_RAW_URL))).willThrow(IOException.class);
+        IOException ioException = mock(IOException.class);
```

Comment:
I was gonna say that may not need to use a mock here, but I then wondered if actually, for an exception it's better to use a mock as it then it may not create a stacktrace 😅